cpp/src/gandiva/precompiled/time_test.cc

```diff
@@ -330,6 +343,19 @@ TEST(TestTime, TimeStampAdd) {
 
   EXPECT_EQ(date_diff_timestamp_int64(StringToTimestamp("2000-02-29 00:00:00"), 365),
             StringToTimestamp("1999-03-01 00:00:00"));
+
+  EXPECT_EQ(
+      subtract_year_month_interval_timestamp(StringToTimestamp("2000-02-27 00:00:00"), 4),
+      StringToTimestamp("1999-10-27 00:00:00"));
```

Comment:
What happens if the user subtracts a value from a `time32` and it results in a negative value? I think it is a good test case to add